frontend/src/metabase/components/MetadataInfo/Popover/Popover.tsx

```diff
@@ -60,14 +58,19 @@ export function Popover({
       transitionProps={{
         duration: group.shouldDelay ? POPOVER_TRANSITION_DURATION : 0,
       }}
-      keepMounted
     >
       <HoverCard.Target>{children}</HoverCard.Target>
       <Dropdown>
         {/* HACK: adds an element between the target and the card */}
         {/* to avoid the card from disappearing */}
         <Target />
-        <WidthBound ref={ref}>{isOpen && content}</WidthBound>
+        <WidthBound
+          ref={node => {
+            setupCloseHandler(node, () => setIsOpen(false));
```

Comment:
Move the `setupCloseHandler` here instead because `handleOpen` will be called before `ref.current` is set. When that happens, if we click inside this hover card, it will still close it because we never call `setupCloseHandler` with the hover card DOM node.